### PR TITLE
Remove the "_precision" suffix from some precision plugin files

### LINKED ISSUE #10021: Inconsistent precision plugin filename suffix
## Proposed refactoring or deprecation

Current names:

```python
pytorch_lightning/plugins/precision/
├── apex_amp.py
├── deepspeed_precision.py
├── double.py
├── fully_sharded_native_amp.py
├── ipu_precision.py
├── mixed.py
├── native_amp.py
├── precision_plugin.py
├── sharded_native_amp.py
├── tpu.py
└── tpu_bfloat.py
```

### Motivation

Had to choose this when working on https://github.com/PyTorchLightning/pytorch-lightning/pull/10020

### Pitch

`ipu_precision.py` -> `ipu.py`
`deepspeed_precision.py` -> `deepspeed.py`

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

<sub>

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, finetuning and solving problems with deep learning

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks and more for research and production with PyTorch Lightning and PyTorch

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

</sub>

### PR DESCRIPTION
**Pull request by puhuk**
Resolve issue #10021

## What does this PR do?

To resolve issue #10021 
Change file names
`ipu_precision.py` -> `ipu.py`
`deepspeed_precision.py` -> `deepspeed.py`

Change code related with those files

Fixes #10021 
## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Yes :)
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by carmocca** [comment]
Hi @puhuk! Thanks for taking care of this

We are currently doing some changes to these files, since this PR moves their location, you will be getting conflicts for a bit.

Thanks for your patience!

**Comment by puhuk** [comment]
1.5 is released.
Then this could be continue :)

**Comment by puhuk** [comment]
@carmocca, is there something I can do with this more?

File: pytorch_lightning/plugins/__init__.py [hunk 1]
```diff
@@ -9,10 +9,10 @@
     TrainingTypePluginsRegistry,
 )
 from pytorch_lightning.plugins.precision.apex_amp import ApexMixedPrecisionPlugin
-from pytorch_lightning.plugins.precision.deepspeed_precision import DeepSpeedPrecisionPlugin
```
Comment (carmocca) [inline pytorch_lightning/plugins/__init__.py:12]: You also need to rename the `deepspeed_precision.py` and `ipu_precision.py` files

**Review by tchaton — APPROVED** [review]
LGTM!

**Review by justusschock — APPROVED** [review]

**Review by carmocca — APPROVED** [review]
